### PR TITLE
cards

### PR DESCRIPTION
**Pull request by omrihecht**
updated cards
- remove mini-employee-card, moved to card-employee
- added social to employee cards
- added option for cover image
- added option for action button instead of menu
- updated design

### CODE REVIEW

File: projects/ui-framework/src/lib/cards/card-add/card-add.component.ts [hunk 1]
```diff
@@ -1,32 +1,21 @@
...
 })
 export class CardAddComponent {
-  constructor() {}
-
-  cardType = CardType;
-
-  @Input() card: AddCardData;
-
+  @Input() card: AddCard;
   @Output() clicked: EventEmitter<void> = new EventEmitter<void>();
```
[the remark this answers is not in the document]
Comment (omrihecht) [reply projects/ui-framework/src/lib/cards/card-add/card-add.component.ts]: ?

File: projects/ui-framework/src/lib/cards/card-add/card-add.interface.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export interface AddCard {
+  title: string;
+  subtitle?: string;
+  action?: (...args: any[]) => void;
```
[the remark this answers is not in the document]
Comment (omrihecht) [reply projects/ui-framework/src/lib/cards/card-add/card-add.interface.ts]: me neither, don't know what it is

File: projects/ui-framework/src/lib/cards/card/card.component.html [hunk 3]
```diff
@@ -1,27 +1,50 @@
...
+   [b-truncate-tooltip]="2"
    [delay]="100"
    [trustCssVars]="true">
-  {{ card.data.text }}
+  {{ card.title }}
 </p>
 
-<div class="card-bottom">
-  <ng-content select="[card-bottom]"></ng-content>
+<div class="card-content">
```
[the remark this answers is not in the document]
Comment (omrihecht) [reply projects/ui-framework/src/lib/cards/card/card.component.html]: cool, didn't know that

File: projects/ui-framework/src/lib/cards/card/card.component.html [hunk 4]
```diff
@@ -1,27 +1,50 @@
...
+  <ng-content select="[card-content]"></ng-content>
+</div>
+
+<div *ngIf="card.footerCtaLabel"
```
[the remark this answers is not in the document]
Comment (omrihecht) [reply projects/ui-framework/src/lib/cards/card/card.component.html]: ok

File: projects/ui-framework/src/lib/cards/card/card.component.scss [hunk 5]
```diff
@@ -42,91 +45,89 @@
...
+    $transform: null,
+    $addCssVars: true
+  );
 }
 
-.card-top,
 .card-content {
-  margin-bottom: 15px;
-  color: $grey-700;
+  margin-bottom: 16px;
```
[the remark this answers is not in the document]
Comment (omrihecht) [reply projects/ui-framework/src/lib/cards/card/card.component.scss]: it always last child, one content element, but i think should always be 0, i need to check

[the remark this answers is not in the document]
Comment (galloween) [reply projects/ui-framework/src/lib/cards/card/card.component.scss]: no, there is sometimes footer after it

File: projects/ui-framework/src/lib/cards/card-employee/card-employee.component.html [hunk 6]
```diff
@@ -1,7 +1,45 @@
...
+            (clicked)="clicked ? onClick($event) : null">
+  </b-avatar>
+
+  <b-display-4 class="title"
+               *bTruncateTooltip="2">
+    {{card.title}}
+  </b-display-4>
+
+  <b-caption class="subtitle"
+             *bTruncateTooltip="2">
```
[the remark this answers is not in the document]
Comment (galloween) [reply projects/ui-framework/src/lib/cards/card-employee/card-employee.component.html]: ah sorry it wont work. you can only use the directive on another component...
(b-caption and [b-truncate-tooltip] are both components)

File: projects/ui-framework/src/lib/cards/card/card.component.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (omrihecht) [reply projects/ui-framework/src/lib/cards/card/card.component.scss]: so i added :empty { display:none }, this solves it

[the remark this answers is not in the document]
Comment (omrihecht) [reply projects/ui-framework/src/lib/cards/card/card.component.scss]: so i am living the margin=0, the space between content and footer is the footer padding, this works well

**Pull request merged**